fly/table_fly_app_certificate.go

```diff
@@ -0,0 +1,129 @@
+package fly
+
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-fly/flyapi"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+
+	provider "github.com/fly-apps/terraform-provider-fly/graphql"
+)
+
+//// TABLE DEFINITION
+
+func tableFlyAppCertificate(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "fly_app_certificate",
+		Description: "Fly App Certificate",
+		List: &plugin.ListConfig{
+			ParentHydrate: listFlyApps,
+			Hydrate:       listFlyAppCertificates,
+		},
+		Get: &plugin.GetConfig{
+			Hydrate:    getFlyAppCertificate,
+			KeyColumns: plugin.SingleColumn("id"),
+		},
+		Columns: []*plugin.Column{
+			{Name: "domain", Description: "The fully qualified domain name of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "id", Description: "A unique identifier of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "hostname", Description: "The hostname of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "created_at", Description: "The timestamp when the certificate was created.", Type: proto.ColumnType_TIMESTAMP},
+			{Name: "source", Description: "The source of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "certificate_authority", Description: "The certificate authority.", Type: proto.ColumnType_STRING},
+			{Name: "verified", Description: "If true, the certificate DNS configuration is verified.", Type: proto.ColumnType_BOOL},
+			{Name: "client_status", Description: "The client status of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "dns_provider", Description: "The DNS provider of the certificate.", Type: proto.ColumnType_STRING},
+			{Name: "dns_validation_hostname", Description: "Specifies the DNS validation hostname.", Type: proto.ColumnType_STRING},
+			{Name: "dns_validation_instructions", Description: "Specifies the DNS validation instructions.", Type: proto.ColumnType_STRING},
+			{Name: "dns_validation_target", Description: "Specifies the DNS validation target.", Type: proto.ColumnType_STRING},
+			{Name: "is_acme_alpn_configured", Description: "If true, certificate acme-alpn is configured.", Type: proto.ColumnType_BOOL},
+			{Name: "is_acme_dns_configured", Description: "If true, acme-dns is configured for domain validation.", Type: proto.ColumnType_BOOL},
+			{Name: "is_apex", Description: "True, if the certificate is a apex certificate.", Type: proto.ColumnType_BOOL},
+			{Name: "is_configured", Description: "If true, certificate is configured with valid DNS configuration.", Type: proto.ColumnType_BOOL},
+			{Name: "is_wildcard", Description: "If true, the hostname of the certificate contains wildcard.", Type: proto.ColumnType_BOOL},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listFlyAppCertificates(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	appData := h.Item.(provider.GetFullAppApp)
+	appID := appData.Name
+
```

Comment:
@misraved, I don't think we need a check to validate the `appID` since if the parentHydrate function returns an empty list, this table will also return the empty list.

Reply:
Thanks for the confirmation @Subhajit97.

Is this a feature of the SK to return an empty result or is it an API behaviour?